.github/workflows/ci.yml

```diff
@@ -9,7 +9,7 @@ jobs:
       - id: checkout
         uses: actions/checkout@v2
       - id: install_dependencies
-        run: brew install ossp-uuid boost flex openssl python lua xz libssh2
+        run: brew install lua@5.3 && brew install ossp-uuid boost flex openssl python xz libssh2
```

Comment:
why do we need two `brew` commands?  `brew install ossp-uuid boost flex openssl python lua@5.3 xz libssh2` doesn't work, or did you just want to make this easier to spot for readers?

Reply:
I managed to make it work when lua@5.3 was installed before the other packages in a single brew command, but then it failed in the next github actions command. I thought the order of install is arbitrary if I use a single brew install command so I separated them, but seems it seems not to be the case, as it is still failing. CMake still finds lua5.4 somehow which needs to be removed.

Reply:
How about 
```yaml
run: brew install ossp-uuid boost flex openssl python lua@5.3 xz libssh2 && brew switch lua 5.3
```

Reply:
It fails with:
```
Error: lua not found in the Cellar.
Error: Process completed with exit code 1.
```

Reply:
I think that means you did not run `brew install lua@5.3` before. If that was run it should by default automatically link it unless a previous version is already present to be found.

Reply:
I'm not sure what the difference is, but in 2 separate runs the same command had 2 different outcomes:
```
Run brew install ossp-uuid boost flex openssl python lua@5.3 xz libssh2 && brew unlink lua && brew link lua@5.3 --force --overwrite
  brew install ossp-uuid boost flex openssl python lua@5.3 xz libssh2 && brew unlink lua && brew link lua@5.3 --force --overwrite
  shell: /bin/bash -e {0}
Warning: openssl@1.1 1.1.1h is already installed and up-to-date
To reinstall 1.1.1h, run `brew reinstall openssl@1.1`
Warning: python@3.9 3.9.0_2 is already installed and up-to-date
To reinstall 3.9.0_2, run `brew reinstall python@3.9`
Warning: xz 5.2.5 is already installed and up-to-date
To reinstall 5.2.5, run `brew reinstall xz`
Warning: libssh2 1.9.0_1 is already installed and up-to-date
To reinstall 1.9.0_1, run `brew reinstall libssh2`
==> Downloading https://homebrew.bintray.com/bottles/ossp-uuid-1.6.2_2.catalina.bottle.tar.gz
==> Pouring ossp-uuid-1.6.2_2.catalina.bottle.tar.gz
🍺  /usr/local/Cellar/ossp-uuid/1.6.2_2: 18 files, 209.5KB
==> Downloading https://homebrew.bintray.com/bottles/boost-1.74.0.catalina.bottle.tar.gz
==> Downloading from https://d29vzk4ow07wi7.cloudfront.net/0c3609787aac666de8db539259e2f604bda631bcc4fde5f5427ea83f4692dec1?response-content-disposition=attachment%3Bfilename%3D%22boost-1.74.0.catalina.bottle.tar.gz%22&Policy=eyJTdGF0ZW1lbnQiOiBbeyJSZXNvdXJjZSI6Imh0dHAqOi8vZDI5dnprNG93MDd3aTcuY2xvdWRmcm9udC5uZXQvMGMzNjA5Nzg3YWFjNjY2ZGU4ZGI1MzkyNTllMmY2MDRiZGE2MzFiY2M0ZmRlNWY1NDI3ZWE4M2Y0NjkyZGVjMT9yZXNwb25zZS1jb250ZW50LWRpc3Bvc2l0aW9uPWF0dGFjaG1lbnQlM0JmaWxlbmFtZSUzRCUyMmJvb3N0LTEuNzQuMC5jYXRhbGluYS5ib3R0bGUudGFyLmd6JTIyIiwiQ29uZGl0aW9uIjp7IkRhdGVMZXNzVGhhbiI6eyJBV1M6RXBvY2hUaW1lIjoxNjA4MTM2MDg5fSwiSXBBZGRyZXNzIjp7IkFXUzpTb3VyY2VJcCI6IjAuMC4wLjAvMCJ9fX1dfQ__&Signature=AO2Et4NcT71qnzbOdzjO2mBhgHX0F~V7NsrZjA0PKHL6Lsr5aVuRPrliKObSKf51AJ8exbhTcIuuNjtyL~AgUCx7ffhOx9~sgECVTrJp9d6fnfTTLgOc0OejJnxmjhkyK0~gwNGUIyPZ5lmvQj4Q1Q2JDmoPNVZNKfKhGwOaA7KM3efSxKK1z2hQCoiFKIpfn660BhpWHUaQ5MLd3U1dPg3k6sFUkZ3IOSU7pcUOzOhLXpNu6iVmHmImFWAnujyQZNE-QfEwH8ePpeDJgMdVBc5Qai9CZiKLmJ9B5atOGK6sH1besmT~BvSAcvQs2LobSYE13dsdvA6nsM-dA4l0vw__&Key-Pair-Id=APKAIFKFWOMXM2UMTSFA
==> Pouring boost-1.74.0.catalina.bottle.tar.gz
🍺  /usr/local/Cellar/boost/1.74.0: 14,695 files, 524.4MB
==> Downloading https://homebrew.bintray.com/bottles/flex-2.6.4_1.catalina.bottle.tar.gz
==> Downloading from https://d29vzk4ow07wi7.cloudfront.net/902e2701bb4d8130fe3177211dda84b6ebc6a520467874a52bcd7ff043b949cc?response-content-disposition=attachment%3Bfilename%3D%22flex-2.6.4_1.catalina.bottle.tar.gz%22&Policy=eyJTdGF0ZW1lbnQiOiBbeyJSZXNvdXJjZSI6Imh0dHAqOi8vZDI5dnprNG93MDd3aTcuY2xvdWRmcm9udC5uZXQvOTAyZTI3MDFiYjRkODEzMGZlMzE3NzIxMWRkYTg0YjZlYmM2YTUyMDQ2Nzg3NGE1MmJjZDdmZjA0M2I5NDljYz9yZXNwb25zZS1jb250ZW50LWRpc3Bvc2l0aW9uPWF0dGFjaG1lbnQlM0JmaWxlbmFtZSUzRCUyMmZsZXgtMi42LjRfMS5jYXRhbGluYS5ib3R0bGUudGFyLmd6JTIyIiwiQ29uZGl0aW9uIjp7IkRhdGVMZXNzVGhhbiI6eyJBV1M6RXBvY2hUaW1lIjoxNjA4MTM2MTIzfSwiSXBBZGRyZXNzIjp7IkFXUzpTb3VyY2VJcCI6IjAuMC4wLjAvMCJ9fX1dfQ__&Signature=Ij8SlZH8abiKyV359dg6m6yb12JfLcNuAx8lOTPZc2V6JqJ9WEctnDbPMYtjjXt4EVREBruQIdYONmKrW6p2MEhdXw0H~qLc0ZJ6NwyyzGOpE5JKj2Qv0dL7ILuc4Q4tx6Hn7zTTqeVvrEfNVen72AU9dHygtJUAx1OhDHgVLI0mCuYHAFSFh7Z3rTNxo55ZGOWoK0b5ABAl0uRdSogj2auMgIPGTvX~CNqx0JC6LDvMX-0FZgzSiStI2Nxl-u0XVw7u5Lyf2-3aoX4CJyhO2Vx-ee~jZS2KsANb4eWnSgIHmvtTWLQTLdXAG1MJMlI8Qv4lSiSnC4HRPcFg3A37Aw__&Key-Pair-Id=APKAIFKFWOMXM2UMTSFA
==> Pouring flex-2.6.4_1.catalina.bottle.tar.gz
==> Caveats
flex is keg-only, which means it was not symlinked into /usr/local,
because macOS already provides this software and installing another version in
parallel can cause all kinds of trouble.

If you need to have flex first in your PATH run:
  echo 'export PATH="/usr/local/opt/flex/bin:$PATH"' >> /Users/runner/.bash_profile

For compilers to find flex you may need to set:
  export LDFLAGS="-L/usr/local/opt/flex/lib"
  export CPPFLAGS="-I/usr/local/opt/flex/include"

==> Summary
🍺  /usr/local/Cellar/flex/2.6.4_1: 45 files, 1.4MB
==> Downloading https://homebrew.bintray.com/bottles/lua-5.3.5_1.catalina.bottle.tar.gz
==> Pouring lua-5.3.5_1.catalina.bottle.tar.gz
==> Caveats
You may also want luarocks:
  brew install luarocks
==> Summary
🍺  /usr/local/Cellar/lua/5.3.5_1: 28 files, 286.6KB
==> Caveats
==> flex
flex is keg-only, which means it was not symlinked into /usr/local,
because macOS already provides this software and installing another version in
parallel can cause all kinds of trouble.

If you need to have flex first in your PATH run:
  echo 'export PATH="/usr/local/opt/flex/bin:$PATH"' >> /Users/runner/.bash_profile

For compilers to find flex you may need to set:
  export LDFLAGS="-L/usr/local/opt/flex/lib"
  export CPPFLAGS="-I/usr/local/opt/flex/include"

==> lua
You may also want luarocks:
  brew install luarocks
Unlinking /usr/local/Cellar/lua/5.3.5_1... 17 symlinks removed
Linking /usr/local/Cellar/lua/5.3.5_1... 17 symlinks created
```

```
Run brew install ossp-uuid boost flex openssl python lua@5.3 xz libssh2 && brew unlink lua && brew link lua@5.3 --force --overwrite
  brew install ossp-uuid boost flex openssl python lua@5.3 xz libssh2 && brew unlink lua && brew link lua@5.3 --force --overwrite
  shell: /bin/bash -e {0}
Warning: openssl@1.1 1.1.1i is already installed and up-to-date
To reinstall 1.1.1i, run `brew reinstall openssl@1.1`
==> Downloading https://homebrew.bintray.com/bottles/ossp-uuid-1.6.2_2.catalina.bottle.tar.gz
Warning: python@3.9 3.9.1 is already installed and up-to-date
==> Pouring ossp-uuid-1.6.2_2.catalina.bottle.tar.gz
To reinstall 3.9.1, run `brew reinstall python@3.9`
Warning: xz 5.2.5 is already installed and up-to-date
To reinstall 5.2.5, run `brew reinstall xz`
Warning: libssh2 1.9.0_1 is already installed and up-to-date
To reinstall 1.9.0_1, run `brew reinstall libssh2`
🍺  /usr/local/Cellar/ossp-uuid/1.6.2_2: 18 files, 209.5KB
==> Downloading https://homebrew.bintray.com/bottles/boost-1.74.0.catalina.bottle.tar.gz
==> Downloading from https://d29vzk4ow07wi7.cloudfront.net/0c3609787aac666de8db539259e2f604bda631bcc4fde5f5427ea83f4692dec1?response-content-disposition=attachment%3Bfilename%3D%22boost-1.74.0.catalina.bottle.tar.gz%22&Policy=eyJTdGF0ZW1lbnQiOiBbeyJSZXNvdXJjZSI6Imh0dHAqOi8vZDI5dnprNG93MDd3aTcuY2xvdWRmcm9udC5uZXQvMGMzNjA5Nzg3YWFjNjY2ZGU4ZGI1MzkyNTllMmY2MDRiZGE2MzFiY2M0ZmRlNWY1NDI3ZWE4M2Y0NjkyZGVjMT9yZXNwb25zZS1jb250ZW50LWRpc3Bvc2l0aW9uPWF0dGFjaG1lbnQlM0JmaWxlbmFtZSUzRCUyMmJvb3N0LTEuNzQuMC5jYXRhbGluYS5ib3R0bGUudGFyLmd6JTIyIiwiQ29uZGl0aW9uIjp7IkRhdGVMZXNzVGhhbiI6eyJBV1M6RXBvY2hUaW1lIjoxNjA4MTM2MDkyfSwiSXBBZGRyZXNzIjp7IkFXUzpTb3VyY2VJcCI6IjAuMC4wLjAvMCJ9fX1dfQ__&Signature=gPN6Ebpwc2Mjr0oJIK8NTUujy73WZ2FsRNDKJkCLRIhkE3a8u1mx42pRY1MurbhYf-xzdOb9KZ~MMFlo2UfxWEzLykURxWM083Q4XQkxD0YiVB52N-GI7VGhi-m9m~OIkDHu2DY60fHXmU6D1BL9~~p242ah-iH5BrO1~VgbIGUugeUz9E6AIC~EkPkrYRFTjvjRDlK7xBLw2jxPHKqHEmTnc1rTvdlzMuj3WV1Sl2nqzx6C5mH~dbiYoZ55l3WOrSvDbXt4IRloAhtR3M7gnj079fPPggRWO-b5yYfWAVJdOblY0OmQx3jXw~ceUZZgMGkkIFFKjajDQaZuKdPvzw__&Key-Pair-Id=APKAIFKFWOMXM2UMTSFA
==> Pouring boost-1.74.0.catalina.bottle.tar.gz
🍺  /usr/local/Cellar/boost/1.74.0: 14,695 files, 524.4MB
==> Downloading https://homebrew.bintray.com/bottles/flex-2.6.4_1.catalina.bottle.tar.gz
==> Downloading from https://d29vzk4ow07wi7.cloudfront.net/902e2701bb4d8130fe3177211dda84b6ebc6a520467874a52bcd7ff043b949cc?response-content-disposition=attachment%3Bfilename%3D%22flex-2.6.4_1.catalina.bottle.tar.gz%22&Policy=eyJTdGF0ZW1lbnQiOiBbeyJSZXNvdXJjZSI6Imh0dHAqOi8vZDI5dnprNG93MDd3aTcuY2xvdWRmcm9udC5uZXQvOTAyZTI3MDFiYjRkODEzMGZlMzE3NzIxMWRkYTg0YjZlYmM2YTUyMDQ2Nzg3NGE1MmJjZDdmZjA0M2I5NDljYz9yZXNwb25zZS1jb250ZW50LWRpc3Bvc2l0aW9uPWF0dGFjaG1lbnQlM0JmaWxlbmFtZSUzRCUyMmZsZXgtMi42LjRfMS5jYXRhbGluYS5ib3R0bGUudGFyLmd6JTIyIiwiQ29uZGl0aW9uIjp7IkRhdGVMZXNzVGhhbiI6eyJBV1M6RXBvY2hUaW1lIjoxNjA4MTM2MTI4fSwiSXBBZGRyZXNzIjp7IkFXUzpTb3VyY2VJcCI6IjAuMC4wLjAvMCJ9fX1dfQ__&Signature=HR2E-TLCiuVmDL1AaTW5OF9xOVMpr8stjeN1hoeVxF0J5MAW7l1QDZ185uV2crZ3VQmdLLNyZ5XnEA59sixbQXWTV0ZS756BgD7YshaYzHdWYNAn7xYU2srrqdwqnsBLG-WZxASszb8Q2kAhiE61EAS05XLS5Z7eedDXGZ-0SQa7RzqXS7dEo414qk4CeJyPR3cXPFQkfnJIUCicmbnBKNljeSIKFNguiRt8NbuqOyrDqfsUBleZAlV5i728indngvtqKt4mR-aVelXEaD2pvf8HjJa8x7kth1uZhi8wRTSNhDc6rvvaeo-LODoxXQoFNn3gvuT~klu4Pz6OXjAEpw__&Key-Pair-Id=APKAIFKFWOMXM2UMTSFA
==> Pouring flex-2.6.4_1.catalina.bottle.tar.gz
==> Caveats
flex is keg-only, which means it was not symlinked into /usr/local,
because macOS already provides this software and installing another version in
parallel can cause all kinds of trouble.

If you need to have flex first in your PATH run:
  echo 'export PATH="/usr/local/opt/flex/bin:$PATH"' >> /Users/runner/.bash_profile

For compilers to find flex you may need to set:
  export LDFLAGS="-L/usr/local/opt/flex/lib"
  export CPPFLAGS="-I/usr/local/opt/flex/include"

==> Summary
🍺  /usr/local/Cellar/flex/2.6.4_1: 45 files, 1.4MB
==> Downloading https://homebrew.bintray.com/bottles/lua%405.3-5.3.6.catalina.bottle.1.tar.gz
==> Pouring lua@5.3-5.3.6.catalina.bottle.1.tar.gz
==> Caveats
You may also want luarocks:
  brew install luarocks

lua@5.3 is keg-only, which means it was not symlinked into /usr/local,
because this is an alternate version of another formula.

If you need to have lua@5.3 first in your PATH run:
  echo 'export PATH="/usr/local/opt/lua@5.3/bin:$PATH"' >> /Users/runner/.bash_profile

For compilers to find lua@5.3 you may need to set:
  export LDFLAGS="-L/usr/local/opt/lua@5.3/lib"
  export CPPFLAGS="-I/usr/local/opt/lua@5.3/include"

For pkg-config to find lua@5.3 you may need to set:
  export PKG_CONFIG_PATH="/usr/local/opt/lua@5.3/lib/pkgconfig"

==> Summary
🍺  /usr/local/Cellar/lua@5.3/5.3.6: 28 files, 367.2KB
==> Caveats
==> flex
flex is keg-only, which means it was not symlinked into /usr/local,
because macOS already provides this software and installing another version in
parallel can cause all kinds of trouble.

If you need to have flex first in your PATH run:
  echo 'export PATH="/usr/local/opt/flex/bin:$PATH"' >> /Users/runner/.bash_profile

For compilers to find flex you may need to set:
  export LDFLAGS="-L/usr/local/opt/flex/lib"
  export CPPFLAGS="-I/usr/local/opt/flex/include"

==> lua@5.3
You may also want luarocks:
  brew install luarocks

lua@5.3 is keg-only, which means it was not symlinked into /usr/local,
because this is an alternate version of another formula.

If you need to have lua@5.3 first in your PATH run:
  echo 'export PATH="/usr/local/opt/lua@5.3/bin:$PATH"' >> /Users/runner/.bash_profile

For compilers to find lua@5.3 you may need to set:
  export LDFLAGS="-L/usr/local/opt/lua@5.3/lib"
  export CPPFLAGS="-I/usr/local/opt/lua@5.3/include"

For pkg-config to find lua@5.3 you may need to set:
  export PKG_CONFIG_PATH="/usr/local/opt/lua@5.3/lib/pkgconfig"

Error: No such keg: /usr/local/Cellar/lua
Error: Process completed with exit code 1.
```

Reply:
It seems like it depends on what VM we get on Github, some of them has Lua pre-installed, some don't.